spring-cloud-data-core/src/main/java/org/springframework/cloud/data/core/StreamDefinition.java

```diff
@@ -16,11 +16,217 @@
 
 package org.springframework.cloud.data.core;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.springframework.core.style.ToStringCreator;
+import org.springframework.util.Assert;
+
+/**
+ * Representation of a defined stream. A stream consists of an
+ * ordered list of modules used to process data. Each module
+ * may contain configuration options provided at the time of
+ * stream creation.
+ * <p>
+ * This stream definition does not include any deployment
+ * or runtime configuration for a stream.
+ *
+ * @see ModuleDefinition
+ *
+ * @author Patrick Peralta
+ */
 public class StreamDefinition {
 
-	private String name;
+	/**
+	 * Name of stream.
+	 */
+	private final String name;
 
+	/**
+	 * DSL definition for stream.
+	 */
+	private final String definition;
```

Comment:
might be a good time to change this field (and the accessor methods) to something like dslText. I'm not crazy about that... could be something better, but I would be happy to never again see: `definition.getDefinition()` :)